iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/IcebergAcidUtil.java

```diff
@@ -243,14 +247,49 @@ public static long getDeleteFilePosition(Record rec) {
     return rec.get(DELETE_FILE_META_COLS.get(MetadataColumns.ROW_POSITION), Long.class);
   }
 
+  private static long hashObjectArray(Object[] values) {
+    Hasher hasher = Hashing.murmur3_128().newHasher();
+
+    for (Object val : values) {
+      if (val == null) {
+        // Unique constant for null
+        hasher.putInt(0xDEADBEEF);
+      } else if (val instanceof Integer) {
+        hasher.putInt((Integer) val);
+      } else if (val instanceof Long) {
+        hasher.putLong((Long) val);
+      } else if (val instanceof String) {
+        hasher.putString((String) val, StandardCharsets.UTF_8);
+      } else if (val instanceof Boolean) {
+        hasher.putBoolean((Boolean) val);
+      } else if (val instanceof Short) {
+        hasher.putShort((Short) val);
+      } else if (val instanceof Byte) {
+        hasher.putByte((Byte) val);
+      } else if (val instanceof Character) {
+        hasher.putChar((Character) val);
+      } else if (val instanceof Double) {
+        hasher.putDouble((Double) val);
+      } else if (val instanceof Float) {
+        hasher.putFloat((Float) val);
+      } else {
+        // Fallback to object's string representation
+        hasher.putLong(Objects.hash(val));
```

Comment:
Consider adding tests for this case since we are using long type for these cases.

Reply:
I updated my fix with a smaller change that also works and this change is no longer part of the fix.